### PR TITLE
Add params of config in create

### LINKED ISSUE #74: Add the variables from the config file as parameters with default values
This will allow a user to run
```
hugie endpoint create --model-name tiiuae/falcon-40b --instance-type g5.12xlarge
```

and get and endpoint with an LLM relatively easily whereas with the config file, it is easy to mess something up.

### PR DESCRIPTION
**Pull request by nsorros**
# Description

This PR allows you to easily deploy models in huggingface inference endpoints by modifying some default params instead of having to construct a valid config json. Among other things we can now deploy an LLM as follows

```
hugie endpoint create \
    --repository tiiuae/falcon-7b \
    --name falcon-7b \
    --instance-type g5.2xlarge \
    --instance-size medium \
    --accelerator gpu \
    --image tgi
```

Fixes #74 

# Checklist

- [x] Tests added
- [x] Relevant issue mentioned
- [x] Documentation updated

### CODE REVIEW

**Comment by nsorros** [comment]
In reality the above command does not work yet as I have not cracked the code of sending valid params for a gpu instance

File: hugie/endpoint.py [hunk 1]
```diff
@@ -84,7 +107,39 @@ def create(
...
+            },
+            "provider": {
+                "vendor": vendor,
+                "region": region,
+            },
+            "type": "protected",
+        }
+        data = InferenceEndpointConfig.parse_obj(data).model_dump()
+    else:
+        data = InferenceEndpointConfig.from_json(data).model_dump()
```
Comment (ivyleavedtoadflax) [inline hugie/endpoint.py:151]: Will this ever get triggered given you have the initial `if not data`?

**Review by ivyleavedtoadflax — APPROVED** [review]

File: hugie/endpoint.py (hunk 1 above)
Comment (nsorros) [reply hugie/endpoint.py:151]: It will if you pass the config only

Comment (ivyleavedtoadflax) [reply hugie/endpoint.py:151]: ahh my bad i didn't see the deleted line :+1:

**Review by ivyleavedtoadflax — COMMENTED** [review]
LGTM. Don't forget the update version number and push to pypi

**Pull request merged**